pkg/object/apirule.go

```diff
@@ -121,7 +121,7 @@ func WithOwnerReference(subs []eventingv1alpha2.Subscription) Option {
 
 // WithRules sets the rules of an APIRule for all Subscriptions for a subscriber.
 func WithRules(certsURL string, subs []eventingv1alpha2.Subscription, svc apigatewayv1beta1.Service,
-	methods ...string,
+	methods ...apigatewayv1beta1.HttpMethod,
```

Comment:
(nit) maybe instead of changing it here, convert the strings to methods when we use this methods, so our rest of the codebase is independent from `apigatewayv1beta1.HttpMethod`. Because we have all the enums in `http` pkg.